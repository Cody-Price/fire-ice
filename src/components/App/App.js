import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { fetchHouses } from '../../thunks/fetchHouses'
import { connect } from 'react-redux'
import Wolf from '../Wolf/Wolf'
import Card from '../Card/Card'

export class App extends Component {

  componentDidMount() {
    this.props.fetchHouses()
  }

  render() {
    const displayedHouses = this.props.houses.map(house => {
      return <Card house={house} />
    })
    return (
      <div className='App'>
        <div className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <h2>Welcome to Westeros</h2>
        </div>
        <div className='Display-info Container'>
          {this.props.isLoading ? <Wolf /> : displayedHouses}
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading,
  houses: state.houses
})

export const mapDispatchToProps = (dispatch) => ({
  fetchHouses: () => dispatch(fetchHouses())
})

export default connect(mapStateToProps, mapDispatchToProps)(App)
