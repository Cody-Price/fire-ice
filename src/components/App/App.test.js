import { App, mapStateToProps, mapDispatchToProps } from './App'
import React from 'react'
import { shallow } from 'enzyme'

describe('App', () => {
  it('should match the snapshot', () => {
    const mockFetchHouses = jest.fn()
    const mockHouses = [{name: 'stark'}, {name: 'tarley'}]
    const wrapper = shallow(<App houses={mockHouses} fetchHouses={mockFetchHouses}/>)
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should return an object with the isLoading bool property', () => {
      // Setup
      const mockHouses = [{name: 'stark'}, {name: 'tarley'}]
      const mockState = {
        isLoading: false,
        houses: mockHouses,
        bogusProperty: true
      }

      const expected = {
        isLoading: false,
        houses: mockHouses
      }

      // Execution
      const mappedProps = mapStateToProps(mockState)

      // Expectation
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a fetchHouses thunk when componentDidMount is called', () => {
      // Setup
      const mockDispatch = jest.fn()

      // Execution
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchHouses()

      // Expectaion
      expect(mockDispatch).toHaveBeenCalled()
    })
  })
})