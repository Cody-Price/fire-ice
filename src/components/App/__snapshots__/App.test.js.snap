// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchHouses={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    houses={
      Array [
        Object {
          "name": "stark",
        },
        Object {
          "name": "tarley",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="App-header"
        >
          <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
          />
          <h2>
            Welcome to Westeros
          </h2>
        </div>,
        <div
          className="Display-info"
        >
          <Card
            house={
              Object {
                "name": "stark",
              }
            }
          />
          <Card
            house={
              Object {
                "name": "tarley",
              }
            }
          />
        </div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />,
            <h2>
              Welcome to Westeros
            </h2>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Westeros",
            },
            "ref": null,
            "rendered": "Welcome to Westeros",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              house={
                Object {
                  "name": "stark",
                }
              }
            />,
            <Card
              house={
                Object {
                  "name": "tarley",
                }
              }
            />,
          ],
          "className": "Display-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "house": Object {
                "name": "stark",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "house": Object {
                "name": "tarley",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="App-header"
          >
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />
            <h2>
              Welcome to Westeros
            </h2>
          </div>,
          <div
            className="Display-info"
          >
            <Card
              house={
                Object {
                  "name": "stark",
                }
              }
            />
            <Card
              house={
                Object {
                  "name": "tarley",
                }
              }
            />
          </div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
              />,
              <h2>
                Welcome to Westeros
              </h2>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to Westeros",
              },
              "ref": null,
              "rendered": "Welcome to Westeros",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Card
                house={
                  Object {
                    "name": "stark",
                  }
                }
              />,
              <Card
                house={
                  Object {
                    "name": "tarley",
                  }
                }
              />,
            ],
            "className": "Display-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "house": Object {
                  "name": "stark",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "house": Object {
                  "name": "tarley",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
